### PR TITLE
Issue #32: Documentation fixes in CONTRIBUTING and fix import in terra-integrated tests

### PR DESCRIPTION
**Pull request by jlatone**
### Summary
Fixed a couple documentation errors I encountered when building.

### Details and comments
Fixed a bad dash character and added cmake flags for installing the terra-integrated tests.

### CODE REVIEW

File: .github/CONTRIBUTING.rst [hunk 1]
```diff
@@ -257,6 +257,8 @@ and then run `unittest` Python framework.
 .. code::
 
   qiskit-aer$ python ./setup.py install
+  # if you had to use --plat-name macosx-10.9-x86_64 for bdist_wheel then you need to do this for install:
+  #   python ./setup.py install -- -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=10.9 -DCMAKE_OSX_ARCHITECTURES:STRING=x86_64
```
[the remark this answers is not in the document]
Comment (atilag) [reply .github/CONTRIBUTING.rst]: Ok, we can`t.

[the remark this answers is not in the document]
Comment (jlatone) [reply .github/CONTRIBUTING.rst]: --plat-name is an option for the `bdist_wheel` subcommand and not others like `install`

**Comment by jlatone** [comment]
@atilag have you ever seen the following json-related error when trying to build aer, more specifically, when Building CXX object CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o:

```
/anaconda3/envs/QiskitDevenv/lib/python3.7/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E __run_co_compile --tidy="/usr/local/bin/clang-tidy;-checks=*;-format-style=google;-header-filter=\"/Users/jlatone/Documents/qiskitdev/qiskit-aer/src\";-quiet" --source=../../../contrib/standalone/qasm_simulator.cpp -- /usr/bin/clang++   -I../../../src -I/usr/local/opt/libomp/include -isystem ../../../src/third-party/headers -stdlib=libc++ -ffast-math -pedantic -Wall -Wfloat-equal -Wundef -Wcast-align -Wwrite-strings -Wmissing-declarations -Wredundant-decls -Wshadow -Woverloaded-virtual -Xpreprocessor -fopenmp -O3 -DNDEBUG -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -mmacosx-version-min=10.9   -std=gnu++14 -MD -MT CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o -MF CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o.d -o CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o -c ../../../contrib/standalone/qasm_simulator.cpp
../../../src/third-party/headers/nlohmann_json.hpp:1201:9: error: static_assert failed "could not find from_json() method in T's namespace" [clang-diagnostic-error]
        static_assert(sizeof(BasicJsonType) == 0,
        ^
```

**Comment by jlatone** [comment]
I also haven't been able to figure out how to build & install aer using `pip`. 
Doing a
`pip install -e .`
runs into the `-std=libc++` issue, and the following didn't work either
`pip install --install-option="--plat-name macosx-10.7-x86_64" -e .`

I'll try other incantations...

**Pull request merged**